### DISCUSSION TITLE
Can Grafana Generic OAuth use Hashicorp Vault as an OIDC provider?

### DISCUSSION THREAD

**Discussion by vtomasr5** [grafana/grafana#65218] [General]
I've set up a Vault cluster and a Grafana deployment. Configured Vault as an OIDC Provider and it works with Boundary, for example. But I can't get it working with Grafana generic oauth.

This is my grafana config:
```
    ...
    [auth.generic_oauth]
    name = Vault
    icon = signin
    enabled = true
    allow_sign_up = true
    auto_login = false
    ;client_id = YOUR_APP_CLIENT_ID            ;INJECTED VIA ENV VAR
    ;client_secret = YOUR_APP_CLIENT_SECRET    ;INJECTED VIA ENV VAR
    scopes = user
    role_attribute_path = grafana-roles
    email_attribute_path = email
    login_attribute_path = email
    ;groups_attribute_path =
    empty_scopes = false
    auth_url = https://vault.example.com/v1/identity/oidc/provider/grafana/authorize
    token_url = https://vault.example.com/v1/identity/oidc/provider/grafana/token
    api_url = https://vault.example.com/v1/identity/oidc/provider/grafana/userinfo
    tls_skip_verify_insecure = true
    tls_client_cert =
    tls_client_key =
    tls_client_ca =
    use_pkce = true
    auth_style = 
    ...
```

When I try to login to Grafana using Vault I get a Permission denied error and I think is because I can't get Grafana to login within Vault first before trying to hit the auth_url (aka authorize) endpoint.

What am I missing in the Grafana config? 🤔

Thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
